### PR TITLE
added nll, softplus, logcosh loss

### PR DESCRIPTION
**Pull request by SubhadityaMukherjee**
I saw Pytorch/TF having some extra loss functions. Thought I would try to add as many as possible to Flux. (This PR has just 3 of them).
They are direct conversions from Python -> Julia. I added the source function in existing libraries as well for comparison. 

### PR Checklist

- [ ] Tests are added
- [ ] Entry in NEWS.md
- [ ] Documentation, if applicable
- [ ] Final review from `@dhairyagandhi96` (for API changes).

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
Thanks for this! Have you seen `softplus` and `logcosh` [here](https://github.com/FluxML/NNlib.jl/blob/d524be7dc1a44c96793c0e9b420469204a85e9e7/src/activation.jl#L1)

**Comment by DhairyaLGandhi** [comment]
It is also already available to Flux after the re-export of NNlib, so it should be in the correct user namespace.

**Comment by SubhadityaMukherjee** [comment]
:)
Ohh. I did a search on this repo and didn’t find it. Haha okay next time I’ll look at the related libraries too. 
On the plus side. It seems like nll wasn’t there. 😄 

Just as a side. If I find something in pytorch/tf that flux doesn’t have yet(IF). I can go ahead and drop a PR if I can make it work right? I’m not sure about the policy here so just confirming. 

Thanks for pointing it out though!

**Comment by DhairyaLGandhi** [comment]
Yes that would be fine. Also, I forget if `nll` was added at some point, worth a quick check.

File: src/losses/functions.jl [hunk 1]
```diff
@@ -7,13 +7,48 @@ Return the loss corresponding to mean absolute error:
...
+
+"""
+    softplus(y,β=1) [Pytorch source](https://pytorch.org/docs/stable/generated/torch.nn.Softplus.html#torch.nn.Softplus)
+
+Return the output of the softplus function with β being a parameter of the softplus
+    
+    (1/β).*(log.(exp.(β .* x).+1))
+"""
+
+softplus(y,β=1) = (1/β).*(log.(exp.(β .* x).+1))
```
Comment (DoktorMike) [inline src/losses/functions.jl:28]: This function defines an input y but refers to an unknown variable x in the exp.

Comment (SubhadityaMukherjee) [reply src/losses/functions.jl]: Oh no.. my bad. I fixed it.

**Comment by CarloLucibello** [comment]
this has been stale for a while and I don't think it's worth having those functions, they are rarely used and simple to implement if needed. Closing this as I'm going through some cleanup of the repo
